include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1114 @@
+            return static_cast<alphabet_type>(*this) == rhs || static_cast<alphabet_type>(*this) < rhs;
+        }
+
+        constexpr bool operator>=(alphabet_type const & rhs) const noexcept
+        {
+            return !(static_cast<alphabet_type>(*this) < rhs);
+        }
+        //!\}
+
+        //NOTE(h-2): If the alphabets ever get ++ -- a.s.o. this proxy should, too.
+    };
+
+    //!\cond
+    //NOTE(h-2): it is entirely unclear to me why we need this
+    template <typename t>
+        requires std::is_same_v<value_type_t<remove_cvref_t<t>>, alphabet_type>
+    static constexpr bool has_same_value_type_v = true;
+    //!\endcond
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Equals the alphabet_type.
+    using value_type        = alphabet_type;
+    //!\brief A proxy type that enables assignment, if the underlying data structure also provides a proxy.
+    using reference         = std::conditional_t<std::is_lvalue_reference_v<reference_t<data_type>>,
+                                                 reference_t<data_type>,
+                                                 reference_proxy_type>;
+    //!\brief Equals the alphabet_type / value_type.
+    using const_reference   = alphabet_type;
+    //!\brief The iterator type of this container (a random access iterator).
+    using iterator          = detail::random_access_iterator<bitcompressed_vector>;
+    //!\brief The const_iterator type of this container (a random access iterator).
+    using const_iterator    = detail::random_access_iterator<bitcompressed_vector const>;
+    //!\brief A signed integer type (usually std::ptrdiff_t)
+    using difference_type   = ranges::difference_type_t<data_type>;
+    //!\brief An unsigned integer type (usually std::size_t)
+    using size_type         = ranges::size_type_t<data_type>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    bitcompressed_vector() = default;
+    constexpr bitcompressed_vector(bitcompressed_vector const &) = default;
+    constexpr bitcompressed_vector(bitcompressed_vector &&) = default;
+    constexpr bitcompressed_vector & operator=(bitcompressed_vector const &) = default;
+    constexpr bitcompressed_vector & operator=(bitcompressed_vector &&) = default;
+    ~bitcompressed_vector() = default;
+
+    /*!\brief Construct/assign from a different range.
+     * \tparam other_range_t The type of range to be inserted; must satisfy seqan3::input_range_concept and
+     *                       seqan3::common_reference_concept<value_type_t<other_range_t>, value_type>.
+     * \param[in]      range The sequences to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `range`.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    template <typename other_range_t>
+    explicit bitcompressed_vector(other_range_t && range)
```

Comment:
This is a constructor so in case of an exception there can not be an half instantiated object and thus the exception clause can be removed from the documentation

Reply:
Formally it may take an lvalue-reference to another object, by spacifying the strong exception guarantee I say that it won't be changed in case an exception is thrown (it won't be changed at all in fact).

Also constructors could change class static variables (say a construction counter) which could be changed in case construction is aborted have way through so I think that we should keep the exceptions specification for constructors in general.

Reply:
Why would the source object ever changed in a construction? I am not hundred percent sure, but I would guess, that constructor would be not well implemented or it is by design. Also for the static variables as you suggest it. We will probably never have this kind of an issue, so I am not seeing a benefit for this. Right now, I would say that constructors are by default strong exception safe unless specifically annotated.

Reply:
Yeah, I agree somewhat, but also we dont have a policy for this, yet. We should collect a list of documention questions somewhere and then discuss and decide them in one session and update the entire codebase after that.
I have left the sections in for now, but they could be removed then, ok?